File: algorithms-coursera-python/multiply_matrices.py
import numpy as np

def main():
    a, b = [], []
    multiply_matrices(a, b)

def multiply_matrices(a, b):
    aRows, aCols = len(a), len(a[0])
    bRows, bCols = len(b), len(b[0])

    if aRows != bRows:
        return None
    
    result = [[0 for _ in range(bCols)] for _ in range(aRows)]
    
    for i in range(aRows):
        for j in range(bCols):
            for k in range(aCols):
                result[i][j] += a[i][k] * b[k][j]

if __name__ == "__main__":
    main()

File: algorithms-coursera-go/multiply_matrices.go
package main

func multiplyMatrices(a, b [][]int) [][]int {
	aRows, aCols := len(a), len(a[0])
	bRows, bCols := len(b), len(b[0])

	var result [][]int

	if aCols != bRows {
		return nil
	}

	for i := 0; i < aRows; i++ {
		for j := 0; j < bCols; j++ {
			for k := 0; k < aCols; k++ {
				result[i][j] += a[i][k] * b[k][j]
			}
		}
	}

	return result
}
